core/core.go

```diff
@@ -226,6 +228,12 @@ func (n *IpfsNode) startOnlineServicesWithHost(ctx context.Context, host p2phost
 	// setup name system
 	n.Namesys = namesys.NewNameSystem(n.Routing)
 
+	// setup ipns republishing
+	n.IpnsRepub = ipnsrp.NewRepublisher(n.Routing, n.Peerstore)
+	n.IpnsRepub.AddName(n.Identity)
+
+	n.Process().Go(n.IpnsRepub.Run)
```

Comment:
Then my function starts to look pretty messy:
```
func (rp *Republisher) Run(p gp.Process) {
    p.Go(func(proc gp.Process) {
        // code 
    }
}
```

which to me seems kinda wonky. My intent is to essentially do (with goprocess) `go n.IpnsRepub.Run(ctx)` which has a fairly simple and readable intent.